test/unittests/state_transition.cpp

```diff
@@ -33,7 +33,11 @@ void state_transition::TearDown()
     auto& selected_vm = trace ? tracing_vm : vm;
 
     /// Optionally enable trace capturing in form of a RAII object.
-    const auto trace_capture = trace ? std::optional<TraceCapture>{std::in_place} : std::nullopt;
+    const auto trace_capture = [trace]() {
```

Comment:
This version uses proper copy elision (mandatory by C++ standard now?). But I guess the version with `.emplace()` should look better now.

Reply:
It kinda looks excessive to use lambda for this simple case. I would prefer to declare `trace_capture` non-const as originally, if no other way.

Reply:
I also prefer simpler version. Although it won't be "functional style".